From ba142cc8aff16b624259705c090701af9e1b4909 Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@denx.de>
Date: Fri, 3 Jul 2020 03:28:13 +0200
Subject: [PATCH 163/164] ARM: dts: stm32: Add AV96 DTO for DH 644-100 DSI
 mezzanine with RPi 7inch display

Add DT overlay for the AV96 board for DH 644-100 mezzanine card with
RPi 7" DSI display attached on top.

Upstream-Status: Inappropriate
Signed-off-by: Marek Vasut <marex@denx.de>
---
 arch/arm/boot/dts/Makefile                    |   1 +
 ...-avenger96-overlay-644-100-x6-rpi7inch.dts | 110 ++++++++++++++++++
 2 files changed, 111 insertions(+)
 create mode 100644 arch/arm/boot/dts/stm32mp157a-avenger96-overlay-644-100-x6-rpi7inch.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 9e7b9a383805..25677855fc57 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -993,6 +993,7 @@ dtb-$(CONFIG_ARCH_STM32) += \
 	stm32mp157a-avenger96.dtb \
 	stm32mp157a-dhcor-avenger96.dtb \
 	stm32mp157a-avenger96-overlay-644-100-x6-otm8009a.dtbo \
+	stm32mp157a-avenger96-overlay-644-100-x6-rpi7inch.dtbo \
 	stm32mp157a-avenger96-overlay-fdcan1-x6.dtbo \
 	stm32mp157a-avenger96-overlay-fdcan2-x6.dtbo \
 	stm32mp157a-avenger96-overlay-i2c1-eeprom-x6.dtbo \
diff --git a/arch/arm/boot/dts/stm32mp157a-avenger96-overlay-644-100-x6-rpi7inch.dts b/arch/arm/boot/dts/stm32mp157a-avenger96-overlay-644-100-x6-rpi7inch.dts
new file mode 100644
index 000000000000..81a382203a9e
--- /dev/null
+++ b/arch/arm/boot/dts/stm32mp157a-avenger96-overlay-644-100-x6-rpi7inch.dts
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) 2020 Marek Vasut <marex@denx.de>
+ */
+#include <dt-bindings/gpio/gpio.h>
+
+/dts-v1/;
+/plugin/;
+
+&{/} {
+	panel {
+		compatible = "powertip,ph800480t013-idf02";
+		power-supply = <&attiny>;
+		backlight = <&attiny>;
+
+		port {
+			panel_in: endpoint {
+				remote-endpoint = <&bridge_out>;
+			};
+		};
+	};
+};
+
+&i2c1 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	touchscreen@38 {
+		compatible = "edt,edt-ft5406";
+		reg = <0x38>;
+		/*
+		 * Disabled, since the IRQ line is not on
+		 * the FPC cable, so we cannot get touch
+		 * IRQs unless its connected otherwise. In
+		 * that case, add entry like this one and
+		 * enable below.
+		 *
+		 * interrupt-parent = <&gpiog>;
+		 * interrupts = <2 IRQ_TYPE_EDGE_FALLING>;
+		 */
+		status = "disabled";
+	};
+
+	attiny: regulator@45 {
+		compatible = "raspberrypi,7inch-touchscreen-panel-regulator";
+		reg = <0x45>;
+	};
+};
+
+&dsi {
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	phy-dsi-supply = <&reg18>;
+	status = "okay";
+
+	bridge@0 {
+		compatible = "toshiba,tc358762";
+		reg = <0>;
+		vddc-supply = <&attiny>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+		status = "okay";
+
+		port@0 {
+			reg = <0>;
+			bridge_in: endpoint {
+				remote-endpoint = <&dsi_out>;
+			};
+		};
+
+		port@1 {
+			reg = <1>;
+			bridge_out: endpoint {
+				remote-endpoint = <&panel_in>;
+			};
+		};
+	};
+
+	ports {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		port@0 {
+			reg = <0>;
+			dsi_in: endpoint {
+				remote-endpoint = <&ltdc_ep1_out>;
+			};
+		};
+
+		port@1 {
+			reg = <1>;
+			dsi_out: endpoint {
+				remote-endpoint = <&bridge_in>;
+			};
+		};
+	};
+};
+
+&ltdc {
+	port {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		ltdc_ep1_out: endpoint@1 {
+			reg = <1>;
+			remote-endpoint = <&dsi_in>;
+		};
+	};
+};
-- 
2.28.0

